File: 2023/01/python/solution.py
from pathlib import Path


with Path(Path(__file__).parent, "../input.txt").open() as f:
    INPUT = f.readlines()

codes = list()

for line in INPUT:
    digits = list()
    line = line.strip()
    for char in line:
        if char.isdigit():
            digits.append(char)
    if len(digits) == 1:
        codes.append(int(digits[0] + digits[0]))
    else:
        codes.append(int(digits[0] + digits[-1]))

print(sum(codes))

File: 2023/01/typescript/solution.ts
import * as fs from "node:fs";

const input = fs.readFileSync("../input.txt", "utf-8").trim();

let codes = [];

for (let line of input.split("\n")) {
  let digits = [];
  for (let char of line.split("")) {
    if (!isNaN(Number(char))) {
      digits.push(char);
    }
  }
  if (digits.length == 1) {
    codes.push(Number(digits[0] + digits[0]));
  } else {
    codes.push(Number(digits[0] + digits[digits.length - 1]));
  }
}
console.log(codes.reduce((acc, curr) => acc + curr, 0));
